ord_schema/validations.py

```diff
@@ -0,0 +1,428 @@
+            # Outcome
+            schema.ReactionWorkup: ValidateReactionWorkup,
+            schema.ReactionOutcome: ValidateReactionOutcome,
+            schema.ReactionProduct: ValidateReactionProduct,
+            schema.Selectivity: ValidateSelectivity,
+            schema.DateTime: ValidateDateTime,
+            schema.ReactionAnalysis: ValidateReactionAnalysis,
+            # Metadata
+            schema.ReactionProvenance: ValidateReactionProvenance,
+            schema.Person: ValidatePerson,
+            # Units
+            schema.Time: ValidateTime,
+            schema.Mass: ValidateMass,
+            schema.Moles: ValidateMoles,
+            schema.Volume: ValidateVolume,
+            schema.Concentration: ValidateConcentration,
+            schema.Pressure: ValidatePressure,
+            schema.Temperature: ValidateTemperature,
+            schema.Current: ValidateCurrent,
+            schema.Voltage: ValidateVoltage,
+            schema.Length: ValidateLength,
+            schema.Wavelength: ValidateWavelength,
+            schema.FlowRate: ValidateFlowRate,
+            schema.Percentage: ValidatePercentage,
+        }[type(message)](message)
+    except KeyError:
+        # NOTE(ccoley): I made the conscious decision to raise an error here,
+        # rather than assume that the message is valid. If a message does not
+        # require any message-level checks (not uncommon), then it should still
+        # be listed in the dictionary switch above withpass. This will force
+        # us to think about what is necessary if/when new messages are added.
+        raise NotImplementedError(f"Don't know how to validate {type(message)}")
+
+
+class ValidationWarning(Warning):
+    pass
+
+def return_message_if_valid(func):
+    def wrapper(message):
+        func(message)
+        return message
+    return wrapper
+
+def ensure_float_nonnegative(message, field):
+    if getattr(message, field) < 0:
+        raise ValueError(f'Field {field} of message '\
+            f'{type(message).DESCRIPTOR.name} must be non-negative')
+
+def ensure_float_range(message, field, min, max):
+    if getattr(message, field) < min or getattr(message, field) > max:
+        raise ValueError(f'Field {field} of message '\
+            f'{type(message).DESCRIPTOR.name} must be between {min} and {max}')
+
+def ensure_units_specified_if_value_defined(message):
+    if message.value: # can't distinguish 0 and unspecified anyway
+        if message.units == message.UNSPECIFIED:
+            raise ValueError(f'Unspecified units for {type(message)} with ' \
+                f'value defined ({message.value})')
+
+def ensure_details_specified_if_type_custom(message):
+    if message.type == message.CUSTOM:
+        if not message.details:
+            raise ValueError(f'Custom type defined for {type(message)}, ' \
+                'but details field is empty')
+
+@return_message_if_valid
+def ValidateReaction(message):
```

Comment:
Please use snake_case for function names; http://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations

Reply:
I'd still like this change.

Reply:
Also missed this original comment for some reason.